collectors/gsuite/gsuite_collector.js

```diff
@@ -59,46 +49,30 @@ class GsuiteCollector extends PawsCollector {
         let collector = this;
 
         const keysEnvVar = process.env.paws_creds;
-        // console.log(process.env.paws_creds);
         if (!keysEnvVar) {
-        throw new Error('The $CREDS environment variable was not found!');
+            throw new Error('The $CREDS environment variable was not found!');
         }
 
         const keys = JSON.parse(keysEnvVar);
         const client = auth.fromJSON(keys);
         client.subject = process.env.paws_email_id;
         client.scopes = ['https://www.googleapis.com/auth/admin.reports.audit.readonly'];
-        // const service = google.admin({version: 'reports_v1', client});
-
 
-
-
-        // let params = state.last_log_id ? {from: state.last_log_id} : {q: "date=[" + state.since + " TO *]", sort: "date:1"};
-        // const collection = auth0Client.getLogs(params);
         let logAcc = [];
-        listLoginEvents(client, function(collection, error){
-            if (error) {
+        let params = {startTime : state.since};
+        listLoginEvents(client, params, function(collection, error){
+            if (error){
                 return callback(error);
             }
-             // console.log('error', error);
-             // console.log('collection', collection);
-            collection.forEach(log => {
-                logAcc.push(log);
-            });
 
-            const nextLogId = (logAcc.length > 0) ? logAcc[logAcc.length-1].log_id : state.last_log_id;
-            const lastLogTs = (logAcc.length > 0) ? logAcc[logAcc.length-1].date : null;
+            const nextLogId = (collection.length > 0) ? collection[collection.length-1].log_id : state.last_log_id;
+            const lastLogTs = (collection.length > 0) ? collection[collection.length-1].id.time : null;
```

Comment:
All API/SDK calls are from previous last log till current last log. So how would we pull duplicate logs?

Reply:
That depends on whether the logs that the API returns arrive in order. I haven't seen any documentation stating whether the events arrive in chronological order. Right now we're assuming that they are chronological, but that might not be the case. 

If we can find documentation stating that they are in chronological order, then the approach of using the last log timestamp as the new start date is valid, otherwise I would prefer we use explicit start and end times, like we do in the okta collector.

Explicit start and end times allows for collection across multiple log types without having to track the last log time of individual log types.